### ISSUE TITLE
eslint does not ignore node_modules in sub projects

### ISSUE THREAD

**Issue by akloeber** [eslint/eslint#5815]
**What version of ESLint are you using?**
2.7.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
rules:
  semi: [1, "always"]
```

**What did you do? Please include the actual source code causing the issue.**
I created the following folder structure:
```
.
├── node_modules
│   ├── eslint
│   │   ├── ...
│   ...
├── package.json
├── index.json
└── sub
    ├── index.js
    ├── node_modules
    │   └── mod
    │       └── mod.js
    └── package.json
```
File `mod.js` contains a missing semicolon:
```
console.log('missing semicolon in module')
```
Then I ran `node_modules/.bin/eslint .` in the root folder.

**What did you expect to happen?**
Modules in `node_modules`of sub projects should be ignored by default too.

**What actually happened? Please include the actual, raw output from ESLint.**
A missing semicolon in `mod.js` is reported as warning:
```
$ npm run lint

> parent@1.0.0 lint dir
> eslint .


.../dir/sub/node_modules/mod/index.js
  3:43  warning  Missing semicolon  semi

✖ 1 problem (0 errors, 1 warning)
```

All modules in the `node_modules` folder of sub project are linted. It can easily be fixed by adding `**/node_modules` to `.eslintignore` but this should be the default instead.

This is just a simple example that points directly to the error. The actual error I ran into looked like the following and it took me some time to find out eslint is trying to lint the `node_modules` folder of the sub project:
```
$ npm run lint

> parent@1.0.0 lint /Users/user/Repos/dir
> eslint sub

Cannot find module 'eslint-config-postcss/es5'
Referenced from: /Users/user/Repos/dir/sub/node_modules/autoprefixer/package.json
Error: Cannot find module 'eslint-config-postcss/es5'
Referenced from: /Users/user/Repos/dir/sub/node_modules/autoprefixer/package.json
    at Object.ModuleResolver.resolve (/Users/user/Repos/dir/node_modules/eslint/lib/util/module-resolver.js:74:19)
    at resolve (/Users/user/Repos/dir/node_modules/eslint/lib/config/config- e.js:475:33)
    at load (/Users/user/Repos/dir/node_modules/eslint/lib/config/config-file.js:492:24)
    at /Users/user/Repos/dir/node_modules/eslint/lib/config/config-file.js:388:36
    at Array.reduceRight (native)
    at applyExtends (/Users/user/Repos/dir/node_modules/eslint/lib/config/config-file.js:365:28)
    at Object.load (/Users/user/Repos/dir/node_modules/eslint/lib/config/config-file.js:527:22)
    at loadConfig (/Users/user/Repos/dir/node_modules/eslint/lib/config.js:67:33)
    at getLocalConfig (/Users/user/Repos/dir/node_modules/eslint/lib/config.js:129:23)
    at Config.getConfig (/Users/user/Repos/dir/node_modules/eslint/lib/config.js:230:22)

npm ERR! Darwin 15.3.0
npm ERR! argv "/Users/user/.nvm/versions/node/v4.2.2/bin/node" "/Users/user/.nvm/versions/node/v4.2.2/bin/npm" "run" "lint"
npm ERR! node v4.2.2
npm ERR! npm  v3.8.3
npm ERR! code ELIFECYCLE
npm ERR! parent@1.0.0 lint: `eslint sub`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the parent@1.0.0 lint script 'eslint sub'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the parent package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint sub
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs parent
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls parent
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/user/Repos/dir/npm-debug.log
```

**Comment by alberto** [comment]
This is by design https://github.com/eslint/eslint/issues/5646
